dotnet/src/Azure.Iot.Operations.Connector/ConnectorTelemetryFactory.cs

```diff
@@ -0,0 +1,151 @@
+using Azure.Iot.Operations.Connector.Models;
+using Azure.Iot.Operations.Connector.Telemetry;
+using Azure.Iot.Operations.Protocol;
+using Azure.Iot.Operations.Protocol.Telemetry;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Factory for creating connector telemetry senders with automatic CloudEvents metadata.
+/// This factory handles the integration between the connector configuration and the base SDK.
+/// </summary>
+public class ConnectorTelemetryFactory
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IMqttPubSubClient _mqttClient;
+    private readonly IPayloadSerializer _serializer;
+    private readonly ConnectorConfiguration _connectorConfig;
+    private readonly CloudEventsMetadataBuilder _metadataBuilder;
+    private readonly ILoggerFactory _loggerFactory;
+    
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ConnectorTelemetryFactory"/> class.
+    /// </summary>
+    /// <param name="applicationContext">The application context from the base SDK.</param>
+    /// <param name="mqttClient">The MQTT pub/sub client from the base SDK.</param>
+    /// <param name="serializer">The payload serializer from the base SDK.</param>
+    /// <param name="connectorConfig">The connector configuration.</param>
+    /// <param name="loggerFactory">Logger factory for creating loggers.</param>
+    public ConnectorTelemetryFactory(
+        ApplicationContext applicationContext,
+        IMqttPubSubClient mqttClient,
+        IPayloadSerializer serializer,
+        ConnectorConfiguration connectorConfig,
+        ILoggerFactory loggerFactory)
+    {
+        _applicationContext = applicationContext ?? throw new ArgumentNullException(nameof(applicationContext));
+        _mqttClient = mqttClient ?? throw new ArgumentNullException(nameof(mqttClient));
+        _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        _connectorConfig = connectorConfig ?? throw new ArgumentNullException(nameof(connectorConfig));
+        _loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+        
+        // Create metadata builder from connector configuration
+        _metadataBuilder = new CloudEventsMetadataBuilder(
+            _connectorConfig.Device,
+            _connectorConfig.Endpoint);
+    }
+    
+    /// <summary>
+    /// Creates a connector telemetry sender for a specific asset and dataset.
+    /// </summary>
+    /// <typeparam name="T">The type of telemetry data to send.</typeparam>
+    /// <param name="asset">Asset identity information.</param>
+    /// <param name="dataSet">Dataset or event group definition.</param>
+    /// <param name="topicPattern">Optional MQTT topic pattern override. If null, uses default pattern.</param>
+    /// <returns>A configured connector telemetry sender.</returns>
+    public Task<ConnectorTelemetrySender<T>> CreateSenderAsync<T>(
```

Comment:
nit: this function probably won't have any async calls in it (no network calls, no semaphores, etc.). You can just make it 

```
public ConnectorTelemetrySender<T> CreateSender<T>(
```